pages/index.js

```diff
@@ -49,6 +49,45 @@ export default function Home() {
 
     const myHeight = ((!isMobile) ? "calc(100vh - 170px)" : "calc(50vh - 85px)");
 
+    useEffect(() => {
+            if(moduleReady) { fetchData(); }
+      }, [moduleReady]);
+
+      useEffect(() => {
+            if(moduleReady){handleUserTabChange("STDOUT"); }
+  }, []);
+
+    async function fetchData() {
+        const url = window.location.search;
+        const gist = "https://gist.githubusercontent.com/";
+        const urlParams = new URLSearchParams(url);
+      
+        if (urlParams.get("code")) {
+            setSourceCode(decodeURIComponent(urlParams.get("code")));
+        } else if (urlParams.get("gist")) {
+            const gistUrl = gist + urlParams.get("gist") + "/raw/";
+            fetch(gistUrl)
+                .then((response) => response.text())
+                .then((data) => {
+                setSourceCode(data);
+                openNotification(
+                    "Source Code loaded from gist.",
+                    "bottomRight"
+                );
+                
+                })
+                .catch((error) => {
+                    console.error("Error fetching data:", error);
+                    openNotification("error fetching .", "bottomRight");
+                    setSourceCode(preinstalled_programs.basic.mandelbrot);
+                });
+        }
+         else {
+             setSourceCode(preinstalled_programs.basic.mandelbrot);
+            }
```

Comment:
```suggestion
                });
        }
```

I think we need not set the source code here, since the default value of `sourceCode` is already `preinstalled_programs.basic.mandelbrot` (which would just remain unchanged if there are no url params provided).

Reply:
Yes, sure.